### ISSUE TITLE
'undefined variable' false positive, kw-arg shadowing same name in outer scope

### ISSUE THREAD

**Issue by erjoalgo** [PyCQA/pylint#1552]
### Steps to reproduce
1.  
```python
some_var = 1
fun = lambda some_var=some_var: some_var+1 # gives error
fun2 = lambda some_var=1: some_var+1 # OK
fun3 = lambda some_var: some_var+1 # OK
fun4 = lambda another_var: some_var+1 # OK
fun5 = lambda another_var: another_var+1 # OK
```
2. pylint

### Current behavior
2nd line gives error

### Expected behavior
no error

### pylint --version output
```
No config file found, using default configuration
pylint 1.6.5, 
astroid 1.4.9
Python 2.7.5 (default, Nov  6 2016, 00:28:07) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

**Comment by PCManticore** [comment]
I can reproduce this bug, thanks!
